dcm4che-net-audit/src/test/java/org/dcm4che3/net/audit/AuditLoggerTest.java

```diff
@@ -0,0 +1,356 @@
+
+        assertThat("auditLoggerInstalled is not as expected", logger.getAuditLoggerInstalled(), is(false));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditRecordRepositoryDevices_WhenCalled() {
+        final Connection auditUDP = new Connection("audit-udp", "localhost", 514);
+        auditUDP.setProtocol(Connection.Protocol.SYSLOG_UDP);
+
+        final AuditRecordRepository arr1 = new AuditRecordRepository();
+        arr1.addConnection(auditUDP);
+
+        final Device arrDevice1 = new Device("syslog");
+        arrDevice1.addDeviceExtension(arr1);
+        arrDevice1.addConnection(auditUDP);
+
+        newLogger.setAuditRecordRepositoryDevices(Collections.singletonList(arrDevice1));
+
+        final Connection auditTCP = new Connection("audit-tcp", "localhost", 516);
+        auditTCP.setProtocol(Connection.Protocol.SYSLOG_TLS);
+
+        final AuditRecordRepository arr2 = new AuditRecordRepository();
+        arr2.addConnection(auditTCP);
+
+        final Device arrDevice2 = new Device("syslog2");
+        arrDevice2.addDeviceExtension(arr2);
+        arrDevice2.addConnection(auditTCP);
+
+        logger.setAuditRecordRepositoryDevices(Collections.singletonList(arrDevice2));
+
+        logger.reconfigure(newLogger);
+
+        final Device toDevice = logger.getAuditRecordRepositoryDevices().get(0);
+        assertThat("auditRecordRepositoryDevices' device name is not as expected", toDevice.getDeviceName(), is(arrDevice1.getDeviceName()));
+        assertThat("auditRecordRepositoryDevices' device extension is not as expected", toDevice.getDeviceExtension(AuditRecordRepository.class), samePropertyValuesAs(arr1));
+        assertThat("auditRecordRepositoryDevices' connections is not as expected", toDevice.getConnections().get(0), samePropertyValuesAs(auditUDP));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditSuppressCriteriaList_WhenCalled() {
+        final AuditSuppressCriteria newCriteria = new AuditSuppressCriteria();
+        newCriteria.setCommonName("NewCommonName");
+        newLogger.setAuditSuppressCriteriaList(Collections.singletonList(newCriteria));
+
+        final AuditSuppressCriteria oldCriteria = new AuditSuppressCriteria();
+        oldCriteria.setCommonName("OldCommentName");
+        logger.setAuditSuppressCriteriaList(Collections.singletonList(oldCriteria));
+
+        logger.reconfigure(newLogger);
+
+        assertThat("auditSuppressCriteriaList is not as expected", logger.getAuditSuppressCriteriaList().get(0), samePropertyValuesAs(newCriteria));
+    }
+
+    @Test
+    public void reconfigure_UpdatesDoIncludeInstanceUID_WhenCalled() {
+        newLogger.setDoIncludeInstanceUID(true);
+        logger.setDoIncludeInstanceUID(false);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("doIncludeInstanceUID is not as expected", logger.getDoIncludeInstanceUID(), is(true));
+    }
+
+    @Test
+    public void reconfigure_UpdatesConnections_WhenCalled() {

```

Comment:
This would be a lot better tested with mock where you simply verify that device.reconfigureConnections(connections, from.connections) was made. So you are not testing Device, that should be done in DeviceTest really.